pkgs/development/compilers/cudatoolkit/redist/overrides.nix

```diff
@@ -10,60 +11,92 @@ in
 
   libcufile = prev.libcufile.overrideAttrs (oldAttrs: {
     buildInputs = oldAttrs.buildInputs ++ [
-      prev.libcublas
+      prev.libcublas.lib
       pkgs.numactl
       pkgs.rdma-core
     ];
     # libcuda needs to be resolved during runtime
-    autoPatchelfIgnoreMissingDeps = true;
+    autoPatchelfIgnoreMissingDeps =
+      ["libcuda.so.1"]
+      # From 12.0 and on, libcufile depends on itself for some reason.
+      ++ lib.optionals (cudaVersionAtLeast "12.0") [
+        "libcufile.so.0"
```

Comment:
Is that an issue? We are adding `$ORIGIN` to the runpaths: https://github.com/NixOS/nixpkgs/blob/885eb7da189344b3186ecba903f6b61ed064c1ab/pkgs/development/compilers/cudatoolkit/redist/build-cuda-redist-package.nix#L45-L47

Reply:
@SomeoneSerge 

I actually got the version constraint backwards? I removed it and tried to build, and the 11.x series failed.

```console
$ for version in 1{1_{4,5,6,7,8},2_{0,1,2}}; do echo "Building cudaPackages_$version.libcufile"; nix build --impure ".#cudaPackages_$version.libcufile" && echo "Success!" || echo "Failure :("; done
Building cudaPackages_11_4.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
error: build of '/nix/store/xj6c1ax6q68d6fb3gr1j9j9kzbsgpkm0-libcufile-1.0.2.10.drv' on 'ssh-ng://connorbaker@nixos-ext.rove-hexatonic.ts.net' failed: builder for '/nix/store/xj6c1ax6q68d6fb3gr1j9j9kzbsgpkm0-libcufile-1.0.2.10.drv' failed with exit code 1;
       last 10 log lines:
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/g7p29455wwpbj10nzf7z6cyrlwqh6hgx-libcufile-1.0.2.10/tools/gdsio
       > warn: auto-patchelf ignoring missing libcuda.so.1 wanted by /nix/store/g7p29455wwpbj10nzf7z6cyrlwqh6hgx-libcufile-1.0.2.10/tools/gdsio
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/g7p29455wwpbj10nzf7z6cyrlwqh6hgx-libcufile-1.0.2.10/tools/gdscheck
       > warn: auto-patchelf ignoring missing libcuda.so.1 wanted by /nix/store/g7p29455wwpbj10nzf7z6cyrlwqh6hgx-libcufile-1.0.2.10/tools/gdscheck
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/x7c4c9fdgzh98q6jb9ixrldlzq69dw0y-libcufile-1.0.2.10-lib/lib/libcufile_rdma.so.1.0.2
       > auto-patchelf failed to find all the required dependencies.
       > Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
       > /nix/store/g7is2hr73nrbmh4ch7j8g78khana65wl-stdenv-linux/setup: line 87: pop_var_context: head of shell_variables not a function context
       > /nix/store/g7is2hr73nrbmh4ch7j8g78khana65wl-stdenv-linux/setup: line 1459: pop_var_context: head of shell_variables not a function context
       > /nix/store/g7is2hr73nrbmh4ch7j8g78khana65wl-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/xj6c1ax6q68d6fb3gr1j9j9kzbsgpkm0-libcufile-1.0.2.10.drv'.
error: builder for '/nix/store/xj6c1ax6q68d6fb3gr1j9j9kzbsgpkm0-libcufile-1.0.2.10.drv' failed with exit code 1
Failure :(
Building cudaPackages_11_5.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
error: build of '/nix/store/3bpx5vrnvaq9w40k58xckaawdi6hc96f-libcufile-1.1.1.25.drv' on 'ssh-ng://connorbaker@nixos-ext.rove-hexatonic.ts.net' failed: builder for '/nix/store/3bpx5vrnvaq9w40k58xckaawdi6hc96f-libcufile-1.1.1.25.drv' failed with exit code 1;
       last 10 log lines:
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/9zwp5s4qa7fyapyz1zq022glqpp2mjv6-libcufile-1.1.1.25/tools/gdsio_verify
       > warn: auto-patchelf ignoring missing libcuda.so.1 wanted by /nix/store/9zwp5s4qa7fyapyz1zq022glqpp2mjv6-libcufile-1.1.1.25/tools/gdsio_verify
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/9zwp5s4qa7fyapyz1zq022glqpp2mjv6-libcufile-1.1.1.25/tools/gds_stats
       > warn: auto-patchelf ignoring missing libcuda.so.1 wanted by /nix/store/9zwp5s4qa7fyapyz1zq022glqpp2mjv6-libcufile-1.1.1.25/tools/gds_stats
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/ajdrqb7776dk1m04aa3m7l71d4w9a997-libcufile-1.1.1.25-lib/lib/libcufile_rdma.so.1.1.1
       > auto-patchelf failed to find all the required dependencies.
       > Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
       > /nix/store/g7is2hr73nrbmh4ch7j8g78khana65wl-stdenv-linux/setup: line 87: pop_var_context: head of shell_variables not a function context
       > /nix/store/g7is2hr73nrbmh4ch7j8g78khana65wl-stdenv-linux/setup: line 1459: pop_var_context: head of shell_variables not a function context
       > /nix/store/g7is2hr73nrbmh4ch7j8g78khana65wl-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/3bpx5vrnvaq9w40k58xckaawdi6hc96f-libcufile-1.1.1.25.drv'.
error: builder for '/nix/store/3bpx5vrnvaq9w40k58xckaawdi6hc96f-libcufile-1.1.1.25.drv' failed with exit code 1
Failure :(
Building cudaPackages_11_6.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
error: build of '/nix/store/i5lisv2ldx7lbjgrxm9z8pmlpl3adj1c-libcufile-1.2.1.4.drv' on 'ssh-ng://connorbaker@nixos-ext.rove-hexatonic.ts.net' failed: builder for '/nix/store/i5lisv2ldx7lbjgrxm9z8pmlpl3adj1c-libcufile-1.2.1.4.drv' failed with exit code 1;
       last 10 log lines:
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/f10v0376i1jcdbwpiczh62yrvfa7vrw6-libcufile-1.2.1.4/tools/gdscp
       > warn: auto-patchelf ignoring missing libcuda.so.1 wanted by /nix/store/f10v0376i1jcdbwpiczh62yrvfa7vrw6-libcufile-1.2.1.4/tools/gdscp
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/f10v0376i1jcdbwpiczh62yrvfa7vrw6-libcufile-1.2.1.4/tools/gdsio
       > warn: auto-patchelf ignoring missing libcuda.so.1 wanted by /nix/store/f10v0376i1jcdbwpiczh62yrvfa7vrw6-libcufile-1.2.1.4/tools/gdsio
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/3vfn2sy5mcvy20c23chgi7yd2q41acrq-libcufile-1.2.1.4-lib/lib/libcufile_rdma.so.1.2.1
       > auto-patchelf failed to find all the required dependencies.
       > Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
       > /nix/store/3l3061gkn21q05fvi2h6fhwnabr0x587-stdenv-linux/setup: line 87: pop_var_context: head of shell_variables not a function context
       > /nix/store/3l3061gkn21q05fvi2h6fhwnabr0x587-stdenv-linux/setup: line 1459: pop_var_context: head of shell_variables not a function context
       > /nix/store/3l3061gkn21q05fvi2h6fhwnabr0x587-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/i5lisv2ldx7lbjgrxm9z8pmlpl3adj1c-libcufile-1.2.1.4.drv'.
error: builder for '/nix/store/i5lisv2ldx7lbjgrxm9z8pmlpl3adj1c-libcufile-1.2.1.4.drv' failed with exit code 1
Failure :(
Building cudaPackages_11_7.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
error: build of '/nix/store/ybvvcdbyprins8485ssw10wg5am6rhc8-libcufile-1.3.0.44.drv' on 'ssh-ng://connorbaker@nixos-ext.rove-hexatonic.ts.net' failed: builder for '/nix/store/ybvvcdbyprins8485ssw10wg5am6rhc8-libcufile-1.3.0.44.drv' failed with exit code 1;
       last 10 log lines:
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/b2jaa6nq8g60q2mpng46bippk6ymfng2-libcufile-1.3.0.44/tools/gdscp
       > warn: auto-patchelf ignoring missing libcuda.so.1 wanted by /nix/store/b2jaa6nq8g60q2mpng46bippk6ymfng2-libcufile-1.3.0.44/tools/gdscp
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/b2jaa6nq8g60q2mpng46bippk6ymfng2-libcufile-1.3.0.44/tools/gdsio
       > warn: auto-patchelf ignoring missing libcuda.so.1 wanted by /nix/store/b2jaa6nq8g60q2mpng46bippk6ymfng2-libcufile-1.3.0.44/tools/gdsio
       > error: auto-patchelf could not satisfy dependency libcufile.so.0 wanted by /nix/store/agcm1vy97jsgj59gghzh7551nd65q4xf-libcufile-1.3.0.44-lib/lib/libcufile_rdma.so.1.3.0
       > auto-patchelf failed to find all the required dependencies.
       > Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
       > /nix/store/3l3061gkn21q05fvi2h6fhwnabr0x587-stdenv-linux/setup: line 87: pop_var_context: head of shell_variables not a function context
       > /nix/store/3l3061gkn21q05fvi2h6fhwnabr0x587-stdenv-linux/setup: line 1459: pop_var_context: head of shell_variables not a function context
       > /nix/store/3l3061gkn21q05fvi2h6fhwnabr0x587-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/ybvvcdbyprins8485ssw10wg5am6rhc8-libcufile-1.3.0.44.drv'.
error: builder for '/nix/store/ybvvcdbyprins8485ssw10wg5am6rhc8-libcufile-1.3.0.44.drv' failed with exit code 1
Failure :(
Building cudaPackages_11_8.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_12_0.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_12_1.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_12_2.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
```

Reply:
Fixed in 99c86613630724dd9e4970e356e7b7fb191241b1

```console
$ for version in 1{1_{4,5,6,7,8},2_{0,1,2}}; do echo "Building cudaPackages_$version.libcufile"; nix build --impure ".#cudaPackages_$version.libcufile" && echo "Success!" || echo "Failure :("; done
Building cudaPackages_11_4.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_11_5.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_11_6.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_11_7.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_11_8.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_12_0.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_12_1.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
Building cudaPackages_12_2.libcufile
warning: Git tree '/home/connorbaker/nixpkgs' is dirty
Success!
```

Reply:
So, we do have to ignore libcufile despite already having `$ORIGIN`?

Reply:
Yes, that’s correct :/

any ideas how to better troubleshoot that or find out ahead of time how to generate these overrides? We could include them as a pass in the CUDA-redist-find-features script.